SqlPersistence/Timeout/SequentialGuid.cs

```diff
@@ -0,0 +1,33 @@
+namespace NServiceBus.Persistence.Sql
+{
+    using System;
+    using System.Threading;
+
+    // From https://github.com/aspnet/EntityFramework/blob/dev/src/Microsoft.EntityFrameworkCore/ValueGeneration/SequentialGuidValueGenerator.cs
+    static class SequentialGuid
+    {
+        static long counter = DateTime.UtcNow.Ticks;
```

Comment:
according to this https://github.com/aspnet/EntityFramework/blob/dev/src/Microsoft.EntityFrameworkCore/ValueGeneration/SequentialGuidValueGenerator.cs#L11

>  Generates sequential <see cref="Guid" /> values using the same algorithm as NEWSEQUENTIALID()

so i think it is safe